lib/app.dart

```diff
@@ -1,11 +1,37 @@
-import 'dart:html' as html;
+import 'package:dashboard/model.dart' as model;
+import 'package:dashboard/navbar/controller.dart' as nav_controller;
 import 'logger.dart';
+import 'dart:html' as html;
+import 'package:dashboard/utils.dart' as utils;
 
 Logger logger = Logger('app.dart');
 
+var links = [
+  model.Link('/charts', 'Charts'),
+  model.Link('/settings', 'Settings'),
+];
+
 class App {
+  nav_controller.Controller navController;
   App() {
-    var body = html.document.getElementsByTagName('body');
-    body.first.append(html.ParagraphElement()..innerText = 'Hello world');
+    navController = nav_controller.Controller(links);
+    _handlePath(null);
+    utils.listenToPathChanges(_handlePath);
+  }
+
+  void _handlePath(html.Event event) {
```

Comment:
Would it make sense to have the navController handle path switching? Just thinking from the perspective that at the moment it doesn't do very much.
Then the app can give the navController not just the links, but the respective controllers that would take over the newly loaded page.
It might add too much indirection though which isn't probably necessary, so perhaps this is more something to keep in mind when adding in the other controllers?

Reply:
The way I'm seeing navbar is just one of the components that route to other pages. We may have a edit settings option under each chart that might take us to the setting page - so I think the main app could hold on to the route handler. Also the firebase login will be in the main app as well.

May be will see how this feels and then move it into navController as a refactor exercise?

Reply:
Added a note about this, will revisit